base/c.jl

```diff
@@ -43,6 +43,29 @@ end
 
 typealias Coff_t FileOffset
 
+# C NUL-terminated string pointers; these can be used in ccall
+# instead of Ptr{Cchar} and Ptr{Cwchar_t}, respectively, to enforce
+# a check for embedded NUL chars in the string (to avoid silent truncation).
+immutable Cstring; p::Ptr{Cchar}; end
+immutable Cwstring; p::Ptr{Cwchar_t}; end
```

Comment:
And
```jl
if box(Int, unbox(Int32, Int.size)) === 8
    bitstype 64 Cstring
    bitstype 64 Cwstring
else
    bitstype 32 Cstring
    bitstype 32 Cwstring
end
```
is giving the mysterious `type definition not allowed inside a local scope` error.